internal/fields/mappings.go

```diff
@@ -0,0 +1,577 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fields
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/multierror"
+)
+
+// CreateValidatorForMappings function creates a validator for the mappings.
+func CreateValidatorForMappings(fieldsParentDir string, opts ...ValidatorOption) (v *Validator, err error) {
+	p := packageRoot{}
+	return createValidatorForMappingsAndPackageRoot(fieldsParentDir, p, opts...)
+}
+
+func createValidatorForMappingsAndPackageRoot(fieldsParentDir string, finder packageRootFinder, opts ...ValidatorOption) (v *Validator, err error) {
+	v = new(Validator)
+	v.injectFieldsOptions.IncludeValidationSettings = false
+	for _, opt := range opts {
+		if err := opt(v); err != nil {
+			return nil, err
+		}
+	}
+
+	fieldsDir := filepath.Join(fieldsParentDir, "fields")
+
+	var fdm *DependencyManager
+	if !v.disabledDependencyManagement {
+		packageRoot, found, err := finder.FindPackageRoot()
+		if err != nil {
+			return nil, fmt.Errorf("can't find package root: %w", err)
+		}
+		if !found {
+			return nil, errors.New("package root not found and dependency management is enabled")
+		}
+		fdm, v.Schema, err = initDependencyManagement(packageRoot, v.specVersion, v.enabledImportAllECSSchema)
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize dependency management: %w", err)
+		}
+	}
+	fields, err := loadFieldsFromDir(fieldsDir, fdm, v.injectFieldsOptions)
+	if err != nil {
+		return nil, fmt.Errorf("can't load fields from directory (path: %s): %w", fieldsDir, err)
+	}
+
+	// Load field definitions from package in a different Validator variable
+	// It allows to check whether or not a given mapping comes from ECS
+	// Or just check if a packages contains some specific field definition (e.g. type array)
+	v.LocalSchema = fields
+	return v, nil
+}
+
+type mappingDefinitions map[string]any
```

Comment:
I'll try to remove this type, to be honest I thought it could help but as you mentioned I've been just converting from one type to another in the code without leveraging the new `mappingDefinitions`.

Reply:
Removed `mappingDefinitions` struct and just used `map[string]any` in this new code.